splunklogger.js

```diff
@@ -162,35 +249,57 @@ SplunkLogger.prototype._initializeConfig = function(config) {
         }
 
         // Take the argument's value, then instance value, then the default value
-        ret.token = config.token || ret.token;
-        ret.name = config.name || ret.name || defaultConfig.name;
-        ret.host = config.host || ret.host || defaultConfig.host;
-        ret.path = config.path || ret.path || defaultConfig.path;
-        ret.protocol = config.protocol || ret.protocol || defaultConfig.protocol;
-        ret.level = config.level || ret.level || defaultConfig.level;
-
-        // Start with the default autoFlush value
-        ret.autoFlush = defaultConfig.autoFlush;
-        // Then check this.config.autoFlush
-        if (this.hasOwnProperty("config") && this.config.hasOwnProperty("autoFlush")) {
-            ret.autoFlush = ret.autoFlush;
-        }
-        // Then check the config.autoFlush, the function argument
-        if (config.hasOwnProperty("autoFlush")) {
-            ret.autoFlush = config.autoFlush;
+        ret.token = utils.orByProp("token", config, ret);
+        ret.name = utils.orByProp("name", config, ret, defaultConfig);
+        ret.level = utils.orByProp("level", config, ret, defaultConfig);
+
+        ret.host = utils.orByProp("host", config, ret, defaultConfig);
+        ret.path = utils.orByProp("path", config, ret, defaultConfig);
+        ret.protocol = utils.orByProp("protocol", config, ret, defaultConfig);
+        ret.port = utils.orByProp("port", config, ret, defaultConfig);
+        ret.port = utils.validateNonNegativeInt(ret.port, "Port");
```

Comment:
Yes, it actually doesn't matter since the string is only used for error messages.